tools/ports/__init__.py

```diff
@@ -271,9 +282,16 @@ def clear_project_build(name):
     port = ports_by_name[name]
     port.clear(Ports, settings, shared)
     build_dir = os.path.join(Ports.get_build_dir(), name)
-    utils.delete_dir(build_dir)
+    if not system_libs.USE_NINJA:
```

Comment:
Please add a comment here explaining why ninja wants to keep the files around.

Reply:
This part of the change is no longer needed since I landed #17857